nevergrad/functions/gym/gym_multi.py

```diff
@@ -0,0 +1,663 @@
+            "deep_stackingmemory_neural": neural_size,
+            "stackingmemory_neural": neural_size,
+            "semideep_stackingmemory_neural": neural_size,
+            "deep_extrapolatestackingmemory_neural": neural_size,
+            "extrapolatestackingmemory_neural": neural_size,
+            "semideep_extrapolatestackingmemory_neural": neural_size,
+            "structured_neural": neural_size,
+            "multi_neural": (min(self.num_time_steps, 50),) + unstructured_neural_size,
+            "noisy_neural": neural_size,
+            "noisy_scrambled_neural": neural_size,
+            "scrambled_neural": neural_size,
+        }
+        shape = shape_dict[control]
+        assert all(
+            c in shape_dict for c in self.controllers
+        ), f"{self.controllers} subset of {shape_dict.keys()}"
+        shape = tuple(map(int, shape))
+        self.policy_shape = shape if "structured" not in control else None
+
+        # Create the parametrization.
+        parametrization = parameter.Array(shape=shape).set_name("ng_default")
+        if "structured" in control and "neural" in control and "multi" not in control:
+            parametrization = parameter.Instrumentation(  # type: ignore
+                parameter.Array(shape=tuple(map(int, self.first_layer_shape))),
+                parameter.Array(shape=tuple(map(int, self.second_layer_shape))),
+            ).set_name("ng_struct")
+        if "conformant" in control:
+            try:
+                if env.action_space.low is not None and env.action_space.high is not None:
+                    low = np.repeat(np.expand_dims(env.action_space.low, 0), self.num_time_steps, axis=0)
+                    high = np.repeat(np.expand_dims(env.action_space.high, 0), self.num_time_steps, axis=0)
+                    init = 0.5 * (low + high)
+                    parametrization = parameter.Array(init=init)
+                    parametrization.set_bounds(low, high)
+            except AttributeError:  # Not all env.action_space have a low and a high.
+                pass
+            if self.subaction_type == int:
+                parametrization.set_integer_casting()
+            parametrization.set_name("conformant")
+
+        # Now initializing.
+        super().__init__(self.gym_multi_function, parametrization=parametrization)
+        self.parametrization.function.deterministic = not ("compiler" in name)
+        self.archive: tp.List[tp.Any] = []
+        self.mean_loss = 0.0
+        self.num_losses = 0
+
+    def evaluation_function(self, *recommendations) -> float:
+        """Averages multiple evaluatioons if necessary."""
+        x = recommendations[0].value
+        if not self.randomized:
+            assert "ompiler" not in self.name
+            return self.gym_multi_function(x, limited_fidelity=False)
+        if "ompiler" not in self.name:
+            return np.sum(self.gym_multi_function(x, limited_fidelity=False) for pb_index in range(23)) / 23.
```

Comment:
Shouldn't gym_multi_function take pb_index as argument there? Is it the case that you don't want to test on uris if compiler is in the name ?

Reply:
We are in the case "ompiler" not in self.name, so this is not compilergym. We are just averaging 23 runs.